apps/studio/src/server/modules/whitelist/whitelist.service.ts

```diff
@@ -0,0 +1,66 @@
+import { TRPCError } from "@trpc/server"
+
+import { db } from "../database"
+
+export const isEmailWhitelisted = async (email: string) => {
+  const lowercaseEmail = email.toLowerCase()
+
+  // Step 1: Check if the exact email address is whitelisted
+  const exactMatch = await db
+    .selectFrom("Whitelist")
+    .where("email", "=", lowercaseEmail)
```

Comment:
is this userinput sanitised?

Reply:
Yes it is sanitised via the zod schema of the caller procedure, but sure added an additional guard here in 805eebf4aa60387f9df7f8d26c09d0c42e876531.